legend-engine-protocol-relational/src/main/java/org/finos/legend/engine/protocol/pure/v1/model/packageableElement/store/relational/connection/specification/DatasourceSpecification.java

```diff
@@ -23,4 +23,9 @@
     public SourceInformation sourceInformation;
 
     public abstract <T> T accept(DatasourceSpecificationVisitor<T> datasourceSpecificationVisitor);
+
+    public String getKey()
```

Comment:
Not very sure we want to bring low-level platform-related logic to protocol classes.
If we follow this approach we can ultimately end up polluting platform classes with various different platform logic like case sensitivity, etc which is currently handled by visitors. I realize this approach might help us with compile-time checks but we probably will end up with too much logic in protocol classes.